feat(i18n, client, curriculum): enable traditional chinese

naomi-lgbt opened the pull request:
Checklist:

<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [X] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [X] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [X] My pull request targets the `main` branch of freeCodeCamp.
- [X] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

<!--If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.-->

Closes #XXXXX

<!-- Feel free to add any additional description of changes below this line -->

This pull request will be implementing the automatic translation tool to convert the Simplified Chinese files into Traditional Chinese files.

- [X] Implement Script - [Example PR](https://github.com/nhcarrigan/actions-testing/pull/30) - [Example Action Run](https://github.com/nhcarrigan/actions-testing/actions/runs/658713671)
  - [X] Merge action into existing workflows, per @RandellDawson's [suggestion](https://github.com/nhcarrigan/actions-testing/pull/27#issuecomment-799920142)?
  - [ ] Auto-request review from freecodecamp/Chinese team to validate parser is working as expected?
- [X] Set up config
- [ ] Prepare deployment pipelines with @raisedadead 
  - [ ] Should deploy on root `freecodecamp.org` domain with `/chinese-traditional` path? ~~`/chinese-classical` path, [mirroring Wikipedia](https://zh-classical.wikipedia.org/wiki/%E7%B6%AD%E5%9F%BA%E5%A4%A7%E5%85%B8:%E5%8D%B7%E9%A6%96) to an extent (discussed this with @QuincyLarson).~~
- [ ] Upload initial files? Or allow action to generate them?

Commit:
feat(client): add chinese classical

Add chinese classical to the all-langs validation and necessary
client build configs.

Commit:
feat(tools): add translation action

Add the action to automatically translate the chinese files and
ship a PR.

Commit:
fix: add files not on crowdin

Add the four client schema files that aren't handled by Crowdin.

RandellDawson commented:
Just a clarification for me.  Will there only be one Chinese language for the site?

nhcarrigan commented:
There will be two - an offering of our hand-translated pages in Simplified Chinese, and a machine-converted Traditional Chinese view.

Commit:
feat(tools): merge workflows

Combine the tranlsate-chinese workflow into the three Crowdin
download workflows.

Commit:
fix(tools): no duplicate install

Don't need to install the tools dependencies twice.

RandellDawson commented:
Then why not call it `chinese-traditional` vs. `chinese-classical`?

nhcarrigan commented:
Because when Quincy and I were discussing where to deploy this, we looked at how Wikipedia does it.

I **thought** they hosted on `zh-classical.wikipedia.org` and that's what I told Quincy, but taking another look that might be "Classical Chinese" which is potentially something different?

I'll switch it back to `chinese-traditional` as I do think that makes more sense.

Commit:
fix: classical -> traditional

Commit:
fix: add remaining files

Need to include these .json files even though they're done through
crowdin or the linter fails.

nhcarrigan commented:
(I'm failing a CI either way 😅 )

Commit:
fix: disable no-unresolved

Disable import/no-unresolved step on the linting of the action
file - otherwise, linter will fail.

RandellDawson commented on config/i18n/all-langs.js:
```diff
@@ -32,7 +34,8 @@ const langDisplayNames = {
   english: 'en-US',
   espanol: 'es-419',
-  chinese: 'zh'
+  chinese: 'zh',
```
I am wondering if this should actually be:

```suggestion
  chinese: 'zh'-CN,
```

nhcarrigan replied:
According to https://www.iana.org/assignments/language-subtag-registry/language-subtag-registry (which comes from the MDN docs), `zh-CN` isn't listed as a valid option. Simplified Chinese would be `zh-Hans` it seems.

RandellDawson commented:
Changing `client/i18n/locales/chinese-traditional/intro.json` is causing the one GitHub action to fail.

nhcarrigan commented:
The problem is if I don't upload that file, then the linter fails because the file doesn't exist. It's a one-off situation, really, since we wouldn't normally have a language we add manually.

Commit:
fix: package-lock

Broke the package-lock.json in the rebase. This commit restores it.

raisedadead committed:
Merge branch 'main' into feat/enable-chinese-classical

Commit:
Merge branch 'main' into feat/enable-chinese-classical

raisedadead commented:
Landed via e46e80e08ffad8f0eb6dee4cf7fd43fdce4d6ba6
